lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -444,12 +444,12 @@ def change_table(model, current_table_name)
             spec_attrs         = spec.schema_attributes(column)
             column_declaration = ::DeclareSchema::Model::Column.new(model, current_table_name, column)
             col_attrs          = column_declaration.schema_attributes
-            if !::DeclareSchema::Model::Column.equivalent_schema_attributes?(spec_attrs, col_attrs)
-              normalized_schema_attributes = spec_attrs.merge(fk_field_options(model, col_name_to_change))
+            normalized_schema_attrs = spec_attrs.merge(fk_field_options(model, col_name_to_change))
```

Comment:
This was a subtle bug: merging the `fk_field_options` needed to happen before checking for equivalence.

Reply:
Nice catch!

Reply:
Hah, well it's a bug I created!